classes/order/Order.php

```diff
@@ -2249,25 +2241,17 @@ public function getOrderInvoiceIdIfHasDelivery()
      * Get warehouse associated to the order.
      *
      * return array List of warehouse
+     *
+     * @deprecated Since 9.0 and will be removed in 10.0
      */
     public function getWarehouseList()
     {
-        $results = Db::getInstance()->executeS(
-            'SELECT id_warehouse
-            FROM `' . _DB_PREFIX_ . 'order_detail`
-            WHERE `id_order` =  ' . (int) $this->id . '
-            GROUP BY id_warehouse'
-        );
-        if (!$results) {
-            return [];
-        }
-
-        $warehouse_list = [];
-        foreach ($results as $row) {
-            $warehouse_list[] = $row['id_warehouse'];
-        }
+        @trigger_error(sprintf(
+            '%s is deprecated since 9.0 and will be removed in 10.0.',
+            __METHOD__
+        ), E_USER_DEPRECATED);
 
-        return $warehouse_list;
+        return [0];
```

Comment:
```suggestion
        return [];
```

Why not return an empty array instead? It seems like it's the equivalent of no warehouses in the DB

Reply:
@jolelievre I chose to return zero because if I am correct, every row of order_detail has a id_warehouse of 0, so it would return an array with zero.

Reply:
Yep, it returned `[0]` before.